pkg/plugin/bundle.go

```diff
@@ -31,31 +31,38 @@ type bundle struct {
 	deprecateWarning         string
 }
 
-type BundleOption func(*bundle)
-
-func WithName(name string) BundleOption {
-	return func(opts *bundle) {
-		opts.name = name
+type BundlePlugin struct {
+	Options struct {
+		name                     string
+		version                  Version
+		plugins                  []Plugin
+		supportedProjectVersions []config.Version
+		deprecateWarning         string
 	}
 }
 
-func WithVersion(version Version) BundleOption {
-	return func(opts *bundle) {
-		opts.version = version
-	}
+func NewBundlePlugin() BundlePlugin {
+	return BundlePlugin{}
```

Comment:
So if `BundlePlugin` is exported, then there is no need of calling `NewBundlePlugin` func. We can make the BundlePlugin not to be exported (bundlePlugin) and have this func.